### PR TITLE
Add API Management Version Sets

### PR DESCRIPTION
**Pull request by darrelmiller**
This checklist is used to make sure that common issues in a pull request are addressed. This will expedite the process of getting your pull request merged and avoid extra work on your part to fix issues discovered during the review process. 

### PR information
- [x] The title of the PR is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).
- [x] Except for special cases involving multiple contributors, the PR is started from a fork of the main repository, not a branch.
- [ ] If applicable, the PR references the bug/issue that it fixes.
- [x] Swagger files are correctly named (e.g. the `api-version` in the path should match the `api-version` in the spec).

### Quality of Swagger
- [x] I have read the [contribution guidelines](https://github.com/Azure/azure-rest-api-specs/blob/master/.github/CONTRIBUTING.md).
- [x] My spec meets the review criteria:
  - [x] The spec conforms to the [Swagger 2.0 specification](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/2.0.md).
  - [x] The spec follows the guidelines described in the [Swagger checklist](../documentation/swagger-checklist.md) document.
  - [x] [Validation tools](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/swagger-checklist.md#validation-tools-for-swagger-checklist) were run on swagger spec(s) and have all been fixed in this PR.

### CODE REVIEW

**Review by anuchandy — CHANGES_REQUESTED** [review]
Please take a look at the model and linter errors

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/340790418

https://travis-ci.org/Azure/azure-rest-api-specs/jobs/340790417

Also it seems swagger is missing definition for userCollection model

**Comment by darrelmiller** [comment]
Hey! @anuchandy I believe I have resolved all the issues with this PR. Thanks!

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json
Comment (anuchandy) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: Note that changing the type of the property `apiVersionSet` in this model is breaking change. 

If we take the diff of `ApiVersionSetContract` (old type) and `ApiVersionSetContractProperties` (new type), it looks like the read-only properties `id`, `name` and `type` won't be available for consumer any more. 

Changing the type is a SDK breaking change. This requires major SDK version upgrade.

Removing properties in the response of a stable api call (without bumping api version) is service side breaking change, which requires ARM approval.

**Comment by anuchandy** [comment]
@ravbhatnagar please take a look at new swagger and breaking change in existing stable api version

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json
Comment (darrelmiller) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: @anuchandy  This isn't breaking a change to the current API, this is fixing an incorrect description.  Customer's who attempt to update those fields in the GA release will not be able to.

In the preview release of this API, the customers could create ApiVersionSet directly when creating the API, but it was decided that the ARM guidelines prevented this from being allowed.  Therefore we had to remove this functionality for our customers in our GA release.  The change to make the fields read-only was missed when doing the update.   Our customers are now waiting for the documentation to be updated so they can use the new ARM compliant way of creating these things in a two step process.

If you could unblock this PR so that we can get that documentation to the customer, it would be most appreciated.

/cc @ravbhatnagar

Comment (anuchandy) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: Also we need ARM review because there is a new swagger `apimversionsets.json`, this is the first PR adding this swagger. So as per the established process, it is required to have this sign-ed off by ARM in this repo. I'm sending an email thread requesting ARM to take a look.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json
Comment (ravbhatnagar) [inline specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json]: @darrelmiller - is this resource type already supported on the service side?

Comment (ravbhatnagar) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json]: The resource type names should be lower camel cased and only contain alphanumeric characters. Which is why I was checking.

**Review by ravbhatnagar — COMMENTED** [review]
Looks good, just one comment

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json
Comment (ravbhatnagar) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: @anuchandy - I think what @darrelmiller is saying is that at GA, customer could not update/assign the value of id, name, type since they were read only. Swagger didnt reflect it. And so now, swagger is updating the description correctly to reflect the service state. If true, this is not a breaking change, since it would have never worked for the customer even if he tried to update.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json
Comment (darrelmiller) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json]: @ravbhatnagar Are you saying that path segment `api-version-sets` should actually be `apiVersionSets` ?  It is currently implemented as the former, but we could change it and make sure will still support the old value as well so we don't break any existing users.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json
Comment (darrelmiller) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: @anuchandy The current SDK is from March 2017 and is a preview https://www.nuget.org/packages/Microsoft.Azure.Management.ApiManagement/3.4.0-preview

Comment (anuchandy) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimapis.json]: @darrelmiller thanks for letting me know. Based on the process we established,  i'll add `potential SDK breaking changes` tag for future reference. 

Once we conclude the below discussion with ARM team, will merge this PR.

File: specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json
Comment (darrelmiller) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json]: @ravbhatnagar  We can do this rename in our 2018-01-01 version which is what we are building the new .net client against.

Comment (ravbhatnagar) [reply specification/apimanagement/resource-manager/Microsoft.ApiManagement/stable/2017-03-01/apimversionsets.json]: @darrelmiller - sounds good. we can target this change in the new api-version.

**Comment by ravbhatnagar** [comment]
Signing off!

**Review by anuchandy — APPROVED** [review]

**Pull request merged**
